### PR TITLE
ObjectTypedictionary added to the deploymentOptions interface to get the object types options and display names 

### PR DESCRIPTION
**Pull request by ssreerama**
This PR contains:
1. The include object type options dictionary that holds Key:OptionsName, Value:DisplayName coming from the DacFx API through the STS
2. DacDeployOptionPropertyStringArray value holds the EnumNames(AKA option name) corresponding to the option display nameThis property is being used for SC and SQL DB project extensions.
3. Also, helps to completely avoid the hardcoded option names in the ADS repo.
4. Corresponding STS PR: https://github.com/microsoft/sqltoolsservice/pull/1574

### CODE REVIEW

File: typings/vscode-mssql.d.ts [hunk 1]
```diff
@@ -499,10 +499,10 @@ declare module 'vscode-mssql' {
 	}
 
 	/**
-	* Interface containing deployment options of integer type, value property holds values from <DacFx>\Product\Source\DeploymentApi\ObjectTypes.cs enum
+	* Interface containing deployment options of string[] type, value property holds enum names from <DacFx>\Product\Source\DeploymentApi\ObjectTypes.cs enum
```
Comment (Charles-Gagnon) [inline typings/vscode-mssql.d.ts:502]: Is it really the enum names? Your description implies it's the display name instead.

Comment (ssreerama) [reply typings/vscode-mssql.d.ts:502]: It holds the enum names, let me update the description clearly :p

**Comment by kisantia** [comment]
can you link the related STS PR for this change?

**Comment by ssreerama** [comment]
Working on DacFx related changes with the team, once they are finalized and completed, will updates this PR accordingly, mostly this PR won't have changes except to get the latest DacFx changes of object types display names

File: typings/vscode-mssql.d.ts [hunk 2]
```diff
@@ -499,10 +499,10 @@ declare module 'vscode-mssql' {
 	}
 
 	/**
-	* Interface containing deployment options of integer type, value property holds values from <DacFx>\Product\Source\DeploymentApi\ObjectTypes.cs enum
+	* Interface containing deployment options of string[] type, value property holds enum names(nothing but option name) from <DacFx>\Product\Source\DeploymentApi\ObjectTypes.cs enum
 	*/
-	export interface DacDeployOptionPropertyObject {
-		value: number[];
+	export interface DacDeployOptionPropertyStringArray {
```
Comment (llali) [inline typings/vscode-mssql.d.ts:504]: why adding the type 'string' in the name? you don't have to rename the interface name

**Review by llali — APPROVED** [review]

File: typings/vscode-mssql.d.ts (hunk 2 above)
Comment (ssreerama) [reply typings/vscode-mssql.d.ts:504]: This interface was created for excludeobjects types only, I used the object and booleans to represent the data types they are holding. So, modified the name to the value datatype!

**Comment by ssreerama** [comment]
@Charles-Gagnon, @llali , @Benjin can you please review this PR and related PRs, I'm planning to send this out by end of this week and have a bugbash ready for team! Thanks :)

**Review by Benjin — APPROVED** [review]

File: typings/vscode-mssql.d.ts (hunk 2 above)
Comment (ssreerama) [reply typings/vscode-mssql.d.ts]: reverted to DacDeployOptionPropertyObject! Thanks

**Review by zijchen — APPROVED** [review]

**Pull request merged**